co-authors-plus.php

```diff
@@ -1211,11 +1197,13 @@ public function search_authors( $search = '', $ignored_authors = array() ) {
 		// Allow users to always filter out certain users if needed (e.g. administrators)
 		$ignored_authors = apply_filters( 'coauthors_edit_ignored_authors', $ignored_authors );
 		foreach ( $found_users as $key => $found_user ) {
-			// Make sure the user is contributor and above (or a custom cap)
 			if ( in_array( $found_user->user_login, $ignored_authors ) ) {
 				unset( $found_users[ $key ] );
+
+			//If user does not have permission, they should not have a term in the first place
```

Comment:
I think this might be mistaken. For example we have an editor and they leave the site so we downgrade them to contributor (or even subscriber). We'd still want them to be co-authors on pieces and not remove them completely from the site and from all the bylines if they can't edit posts anymore.

I'm also not sure a change like this should be bundled with this PR. It feels a bit like there was another potential issue that was found while working on the main issue and the fix gets tagged along. Creating multiple issues might be better in the long run.

Reply:
> we have an editor and they leave the site so we downgrade them to contributor

Thanks for bringing this up, I hadn't thought about it! I still think this change would be a good one to remove a lot of garbage from the database, especially for large sites with lots of subscribers. Would also checking whether the user is coauthor of any pieces fix it? 

> It feels a bit like there was another potential issue that was found while working on the main issue and the fix gets tagged along

At first, I had actually completely removed that check, thinking that after my changes all users who had a term were good enough to be displayed, since I was creating terms only for relevant users. Of course, I hadn't thought about existing terms created in previous plugin version, so I restored that back. And then I also thought: "my new feature is going to leave databases quite clear, but only on new installations. Can I do something for old ones as well?". So that's its story, just to let you know what brought me to edit that piece - but I can certainly split it in a different PR if you believe would be more fit :)